benchmark: increase process priority

yaacovCR opened the pull request:
...to try to avoid involuntary context switches.

This seems to me to be pretty TETAW ("too easy to actually work").

yaacovCR committed:
benchmark: increase process priority

to try to avoid involuntary context switches

yaacovCR commented:
@github-actions run-benchmark

yaacovCR commented:
Not sure if sudo can help us here. if i were github actions, it wouldn't!

yaacovCR commented:
@github-actions run-benchmark

yaacovCR commented:
@github-actions run-benchmark

yaacovCR commented:
@github-actions run-benchmark

yaacovCR commented:
@github-actions run-benchmark

yaacovCR commented:
sudo and setPriority seem to be working! although another erupts when writing for some reason...

I can read more about affinity

yaacovCR commented:
I am guessing when using a shell, piping via 3 does not work

https://github.com/graphql/graphql-js/blob/b91f51b8f9c6bc7a2d0b5d1ef7b6121bf094cac4/resources/benchmark.ts#L408

IvanGoncharov commented:
@yaacovCR You need to pass `sudo -C4` 
https://superuser.com/questions/394522/sudo-preserve-non-std-file-descriptors

IvanGoncharov commented on resources/benchmark.ts:
```diff
@@ -376,9 +376,12 @@ interface BenchmarkSample {
     import { benchmark } from '${modulePath}';
 
+    os.setPriority(os.constants.priority.PRIORITY_HIGHEST);
+
```
@yaacovCR Also, you can try setting priority for benchmark itself to the lowest.
That will force context switches to occur before/after measurments.

yaacovCR replied:
Or maybe benchmark will never finish? I think we can see what happens with this as a first step and take it from there.

yaacovCR commented:
> @yaacovCR Simpler solution would be to run the entire benchmark under sudo.

That works for me.

As a side note, I want to make sure this works cross platform. I would assume that ci would add sudo to the whole command and local users would be required to do so on their own or use the windows equivalent.

yaacovCR commented:
@github-actions run-benchmark

yaacovCR committed:
try manually passing along the uid

yaacovCR commented:
@github-actions run-benchmark

yaacovCR committed:
try inner sudo

@IvanGoncharov sudo on the benchmark was not working did not seem to pass through

yaacovCR commented:
@IvanGoncharov 

> sudo: you are not permitted to use the -C option

Needs further research

IvanGoncharov commented:
> Needs further research

@yaacovCR Can I quickly try to run some experiments on top of your PR?

yaacovCR commented:
Closing for now. Would be on wishlist to revisit in terms of our overall benchmarking picture.
